File: api-asn/src/tr/gov/tubitak/uekae/esya/api/asn/cms/ESignaturePolicy.java
package tr.gov.tubitak.uekae.esya.api.asn.cms;

import tr.gov.tubitak.uekae.esya.api.asn.BaseASNWrapper;
import tr.gov.tubitak.uekae.esya.api.common.ESYAException;
import tr.gov.tubitak.uekae.esya.asn.cms.SignaturePolicy;
import tr.gov.tubitak.uekae.esya.asn.cms.SignaturePolicyId;

public class ESignaturePolicy extends BaseASNWrapper<SignaturePolicy>
{
	public ESignaturePolicy(SignaturePolicy aObject)
	{
		super(aObject);
	}
	
	public ESignaturePolicy(byte[] aBytes) throws ESYAException
	{
		super(aBytes,new SignaturePolicy());
	}

    public ESignaturePolicy(ESignaturePolicyId policyId){
        super(new SignaturePolicy());
        mObject.set_signaturePolicyId(policyId.getObject());
    }

	public ESignaturePolicyId getSignaturePolicyId()
	{
		if(mObject.getChoiceID()==SignaturePolicy._SIGNATUREPOLICYID)
			return new ESignaturePolicyId((SignaturePolicyId)mObject.getElement());

		return null;
	}
}

File: api-asn/csharp/src/api/asn/cms/ESignaturePolicy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using tr.gov.tubitak.uekae.esya.api.asn;
using tr.gov.tubitak.uekae.esya.asn.cms;
using tr.gov.tubitak.uekae.esya.api.asn.cms;

namespace tr.gov.tubitak.uekae.esya.api.asn.cms
{
    public class ESignaturePolicy : BaseASNWrapper<SignaturePolicy>
    {
        public ESignaturePolicy(SignaturePolicy aObject)
            : base(aObject)
        {

        }

        public ESignaturePolicy(byte[] aBytes)
            : base(aBytes, new SignaturePolicy())
        {

        }
        public ESignaturePolicy(ESignaturePolicyId policyId)
            : base(new SignaturePolicy())
        {
            mObject.Set_signaturePolicyId(policyId.getObject());
        }
        public ESignaturePolicyId getSignaturePolicyId()
        {
            if (mObject.ChoiceID == SignaturePolicy._SIGNATUREPOLICYID)
                return new ESignaturePolicyId((SignaturePolicyId)mObject.GetElement());

            return null;
        }
    }
}
